src/dash-playlist-loader.js

```diff
@@ -403,52 +420,10 @@ export default class DashPlaylistLoader extends EventTarget {
     if (this.media_) {
       this.trigger('mediachanging');
     }
-
-    if (!playlist.sidx) {
-      // Continue asynchronously if there is no sidx
-      // wait one tick to allow haveMaster to run first on a child loader
-      this.mediaRequest_ = window.setTimeout(
-        this.haveMetadata.bind(this, { startingState, playlist }),
-        0
-      );
-
-      // exit early and don't do sidx work
-      return;
-    }
-
-    // sidxMapping is used when parsing the masterXml, so store
-    // it on the masterPlaylistLoader
-    const oldMaster = this.masterPlaylistLoader_.master;
-    const sidxMapping = this.masterPlaylistLoader_.sidxMapping_;
-    const sidxKey = generateSidxKey(playlist.sidx);
-
-    sidxMapping[sidxKey] = {
-      sidxInfo: playlist.sidx
-    };
-
-    this.request = requestSidx_(
-      this,
-      playlist.sidx,
-      playlist,
-      this.vhs_.xhr,
-      { handleManifestRedirects: this.handleManifestRedirects },
-      this.sidxRequestFinished_(playlist, oldMaster, startingState, (newMaster, sidx) => {
-        if (!newMaster || !sidx) {
-          throw new Error('failed to request sidx');
-        }
-
-        // update loader's sidxMapping with parsed sidx box
-        sidxMapping[sidxKey].sidx = sidx;
-
-        addSidxSegmentsToPlaylist(playlist, sidx, playlist.sidx.resolvedUri);
-
-        // everything is ready just continue to haveMetadata
-        this.haveMetadata({
-          startingState,
-          playlist
-        });
-      })
-    );
+    this.addSidxSegments_(playlist, startingState, (sidxChanged) => {
```

Comment:
Since this function now calls back for playlists that don't have sidx segments, we don't have to separate the logic.